src/lemlib/util.cpp

```diff
@@ -54,44 +53,34 @@ int sgn(float x) {
     else return 1;
 }
 
-/**
- * @brief Return the average of a vector of numbers
- *
- * @param values
- * @return float
- */
-float avg(std::vector<float> values) {
-    float sum = 0;
-    for (float value : values) { sum += value; }
-    return sum / values.size();
-}
-
 /**
  * @brief Exponential moving average
  *
  * @param current current measurement
  * @param previous previous output
  * @param smooth smoothing factor (0-1). 1 means no smoothing, 0 means no change
- * @return float - the smoothed output
+ * @return the smoothed output
  */
-float ema(float current, float previous, float smooth) { return (current * smooth) + (previous * (1 - smooth)); }
+template <isQuantity Q> Q ema(Q current, Q previous, float smooth) {
```

Comment:
shouldnt this be in the header file? If you want a global template function it has to be defined in a header file